website/docs/basics/architecture-concepts.md

```diff
@@ -0,0 +1,248 @@
+- **Theme** - Used in the theme (frontend) part of your project. You can find the files inside the `assets` folder with no suffix. You would use these files for stuff related to your project that is not associated with components and blocks. These files are rarely used, as styles and JavaScript functionality are usually provided at component, block or wrapper levels.
+
+Using any of these classes is optional, and you can use only what you need. The `wp boilerplate setup_theme` command will set up all of the three classes already implemented. Keep in mind that Enqueue classes work in combination with the Webpack build of your project.
+
+All of the asset files mentioned are prefixed with `application`. If you don't like the prefix, you can change it by providing project overrides for constants used in Eightshift Libs. If you change the file names, you must make changes to the Webpack build process as well. For modifying the Webpack configuration, read the [Webpack chapter](webpack).
+
+#### Exception
+
+This namespace contains all exceptions that can be thrown from the project.
+
+#### Helpers
+
+All the backend helpers provided by Eightshift Libs are part of the Helpers namespace. Generally, you can use them from Eightshift Libs without extending or modifying them.
+
+#### I18n
+
+Classes in this namespace are used to define all hooks related to internationalization and localization. For more information about i18n and l10n, check out [Tips and tricks](tips-tricks).
+
+#### Main
+
+The Main namespace contains the `Main` class, which sets up the dependency injection container and autowiring using `AbstractMain` and `Autowiring` classes. For most projects, `Main` can be used automagically without any work on your part, but as with all of the Eightshift Development Kit, you can extend it to suit your needs.
+
+You can provide manual service classes here as well. Learn more in the [autowiring chapter](autowiring).
+
+#### Manifest
+
+The Manifest namespace provides the `manifest.json` file location and helpers to return the full path for a specific asset. Learn more about the manifest in the [manifest chapter](manifest).
+
+#### Media
+
+The Media namespace is used to add custom implementations for media, such as adding custom image sizes, enabling SVG image support, etc. By default, it registers theme support for title tags, HTML5 and post thumbnails.
+
+#### Menu
+
+This class is used to register menu positions.
+
+#### ModifyAdminAppearance
+
+This class is used to change admin appearance. We use it to change the admin color palette depending on the environment.
+
+#### Plugin
+
+This namespace is used only in the plugin boilerplate and holds classes required for activation and deactivation of the plugin.
```

Comment:
Not sure this is correct 🤔 

I know that in the themes we used this to override some specific plugin's functionality (like ACF modifying dropdowns, or GF modifying their built-in block attributes to be compatible with our blocks).

Reply:
Thanks for flagging all of these concerns.

The namespace descriptions are mostly taken from existing content. However, I've went ahead and fact-checked them while writing this. In eightshift-libs, the `Plugin` namespace is only used for plugin (de)activation: https://github.com/infinum/eightshift-libs/tree/develop/src/Plugin

We do use the `Plugins` (note the plural form) namespace on projects to integrate with external plugins (think ACF field registration). This isn't part of eightshift-libs. 

I believe that this is correct given the state of eightshift-libs, but not for all Eightshift-based projects. Goes to show how extensible the development kit is when the maintainers forget something is a custom implementation ;)